src/server/common/model/answer/origin-type/origin-type.js

```diff
@@ -1,26 +1,53 @@
 import { RadioButtonAnswer } from '../radio-button/radio-button.js'
-/** @import {RadioButtonConfig} from '../radio-button/radio-button.js' */
+/** @import {RadioButtonConfig, RadioButtonConfigFactory} from '../radio-button/radio-button.js' */
+/** @import {RawApplicationState} from '~/src/server/common/model/state/state-manager.js' */
 
 /**
  * export @typedef {'tb-restricted-farm' | 'afu' | 'other'} OriginTypeData
  * @typedef {{ originType: OriginTypeData }} OriginTypePayload
  */
 
+/**
+ * @param {RawApplicationState} app
+ * @returns boolean
+ */
+const isOnToTheFarm = (app) => app.origin?.onOffFarm === 'on'
+
+const tbRestrictedOption = { label: 'TB restricted farm' }
+const afuOption = {
+  label: 'Approved finishing unit (AFU)',
+  hint: 'Including enhanced with grazing (AFUE)'
+}
+
+const offFarmOptions = {
+  'tb-restricted-farm': tbRestrictedOption,
+  afu: afuOption,
+  other: { label: 'Another type of premises' }
+}
+
+const onFarmOptions = {
+  market: { label: 'Market' },
+  'unrestricted-farm': {
+    label: 'Unrestricted farm or premises'
+  },
+  'tb-restricted-farm': tbRestrictedOption,
+  afu: afuOption,
+  zoo: { label: 'Zoo' },
+  lab: { label: 'Laboratory' },
+  'after-import-location': {
+    label: 'Location after animals have been imported'
+  },
+  other: { label: 'Another origin' }
+}
+
 /** @augments {RadioButtonAnswer<OriginTypePayload>} */
 export class OriginTypeAnswer extends RadioButtonAnswer {
-  /** @type {RadioButtonConfig} */
-  static config = {
+  /** @type {RadioButtonConfigFactory} */
+  static config = (app) => ({
```

Comment:
I see what you're trying to do
but I disagree that this should be a static function, it is using the state, effectively of the application, and thus should be done within the answer, which would mean it could use `this.value` rather than any complications to pass context in

Reply:
as typified by adding the `new` keyword to the tests

Reply:
even if you need to pass app state in to base result on other sections, i feel it should still be instatiated and not static